Octokit/Http/Connection.cs

```diff
@@ -644,10 +647,22 @@ internal static TwoFactorType ParseTwoFactorType(IResponse restResponse)
 
         static string FormatUserAgent(ProductHeaderValue productInformation)
         {
+            var format =
+#if !HAS_ENVIRONMENT
+                "{0} ({1}; {2}; {3}; Octokit {4})";
+#else
+                "{0} ({1} {2}; {3}; {4}; Octokit {5})";
+#endif
+
             return string.Format(CultureInfo.InvariantCulture,
+                format,
+#if !HAS_ENVIRONMENT
+                productInformation,
+                RuntimeInformation.OSDescription,
+                RuntimeInformation.OSArchitecture.ToString().ToLowerInvariant(),
+#elif NETFX_CORE
                 "{0} ({1} {2}; {3}; {4}; Octokit {5})",
```

Comment:
since `format` is being set above and included in all cases, I think this line may be spurious?

infact do we even really need this hardcoded WindowsRT/Windows8 etc stuff below @shiftkey? 

Can this `NETFX_CORE` case here just be removed and either user `Environment` or `RuntimeInformation` according to the `HAS_ENVIRONMENT` preprocessor definition now added?